### PR TITLE
statistics: Add benchmark for BuildHistAndTopN

### LINKED ISSUE #48699: add benchmark for BuildHistAndTopN
## Enhancement

TiDB has no benchmark for BuildHistAndTopN. we need it to verify the future improvement.

### PR DESCRIPTION
**Pull request by Zheaoli**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/48699

Problem Summary:

### What is changed and how it works?

Fix #48699

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [x] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/statistics/builder_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
+// Copyright 2017 PingCAP, Inc.
```
Comment (hawkingrei) [inline pkg/statistics/builder_test.go:1]:
```suggestion
// Copyright 2023 PingCAP, Inc.
```

File: pkg/statistics/builder_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"github.com/pingcap/tidb/pkg/parser/mysql"
+	"github.com/pingcap/tidb/pkg/types"
+	"github.com/pingcap/tidb/pkg/util/memory"
+	"github.com/pingcap/tidb/pkg/util/mock"
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func BenchmarkBuildHistAndTopN(b *testing.B) {
+	b.ResetTimer()
```
Comment (hawkingrei) [inline pkg/statistics/builder_test.go:27]: We should ```ResetTimer``` after preparing the data.

**Review by hawkingrei — APPROVED** [review]

**Review by winoros — COMMENTED** [review]
It's better to create a dataset with 100K rows which is the more common scenario

**Review by winoros — APPROVED** [review]

**Comment by hawkingrei** [comment]
/release-7.5

**Pull request merged**